internal/controllers/suite_test.go

```diff
@@ -36,14 +35,18 @@ import (
 )
 
 type testingContext struct {
-	env       *envtest.Environment
-	done      chan struct{}
+	env *envtest.Environment
+	context.Context
 	k8sClient client.Client
 }
 
-func setup(reader templates.Reader, helm Helm, objects []runtime.Object) (*testingContext, error) {
+var cancel context.CancelFunc
```

Comment:
what do you think if we return `cancel` as part of `*testingContext` to avoid using global vars?

```golang

type testingContext struct {
	env *envtest.Environment
	cancel context.CancelFunc
	context.Context
	k8sClient client.Client
}

func setup(reader templates.Reader, helm Helm, objects []client.Object) (*testingContext, error) {
	cancelCtx, cancel := context.WithCancel(context.Background())

	ctx := &testingContext{
		cancel: cancel,
		Context: cancelCtx,
		env: &envtest.Environment{
			CRDDirectoryPaths: []string{filepath.Join("..", "..", "config", "crd", "bases")},
		},
	}

}
```

Reply:
Great call. I'm embarrassed I didn't see it.